### PR TITLE
docs: describe building and testing makefiles in DEVELOPER.md

### PR DESCRIPTION
**Pull request by w-bonelli**
Expand documentation on building/testing makefiles

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -432,6 +432,24 @@ Tests should ideally follow a few conventions for easier maintenance:
 
 **Note:** If all three external model repositories are not installed as described above, some tests will be skipped. The full test suite includes >750 cases. All must pass before changes can be merged into this repository.
 
+## Generating makefiles
+
+Run `build_makefiles.py` in the `distribution/` directory after adding, removing, or renaming source files.
+
+If the utilities in `utils` require a new module from the modflow6 `src/` directory, add it to `make/extrafiles.txt`.
```
Comment (emorway-usgs) [inline DEVELOPER.md:439]:
```suggestion
If the utilities located in the `utils` directory (e.g., `mf5to6`) require a new module added to the modflow6 `src/` directory, add it to the respective utility's `make/extrafiles.txt` file.
```

Comment (w-bonelli) [reply DEVELOPER.md]: committed suggestion

File: DEVELOPER.md [hunk 2]
```diff
@@ -432,11 +436,59 @@ Tests should ideally follow a few conventions for easier maintenance:
...
+
+Run `build_makefiles.py` in the `distribution/` directory after adding, removing, or renaming source files. This script uses [Pymake](https://github.com/modflowpy/pymake) to regenerate makefiles. For instance:
+
+```shell
+python build_makefiles.py
+```
+
+### Updating extra and excluded files
+
+If the utilities located in the `utils` directory (e.g., `mf5to6` and `zbud6`) require a new module that was recently added to the modflow6 `src/` directory, then the new module source file should also be added to the utility's `utils/<util>/pymake/extrafiles.txt` file. This file informs Pymake of source files living outside the main source directory, so they can be included in generated makefiles.
```
Comment (langevin-usgs) [inline DEVELOPER.md:449]:
```suggestion
If the utilities located in the `utils` directory (e.g., `mf5to6` and `zbud6`) are affected by changes to the modflow6 `src/` directory (such as new or refactored source files), then the new module source file should also be added to the utility's `utils/<util>/pymake/extrafiles.txt` file. This file informs Pymake of source files living outside the main source directory, so they can be included in generated makefiles.
```

File: DEVELOPER.md [hunk 3]
```diff
@@ -432,11 +436,59 @@ Tests should ideally follow a few conventions for easier maintenance:
...
+
+Module dependencies for features still under development should be added to `excludefiles.txt`. Souce files listed in this file will be excluded from makefiles generated by Pymake. Makefiles should only include source files for officially released/supported features.
```
Comment (langevin-usgs) [inline DEVELOPER.md:451]:
```suggestion
Module dependencies for features still under development should be added to `excludefiles.txt`. Souce files listed in this file will be excluded from makefiles generated by Pymake. Makefiles should only include the source files needed to the build officially released/supported features.
```

**Review by langevin-usgs — COMMENTED** [review]
Nice.

**Pull request merged**